docs/learn/expert-guides/set-token-metadata-on-deploy.md

```diff
@@ -0,0 +1,191 @@
+  const tokenBytecodeWithConstructor = ethers.concat([
+    tokenBytecode,
+    encodedConstructorParams,
+  ]);
+
+  // Get the address of the custom token contract that will be created
+  const customTokenAddress = await universalProfile.execute.staticCall(
+    1, // Operation type: CREATE
+    ethers.ZeroAddress, // Target: 0x0 as contract will be initialized
+    0, // Value is empty
+    tokenBytecodeWithConstructor, // Payload of the contract
+  );
+
+  // Encode the metadata for deployment
+  const encodedLSP4Metadata = ERC725.encodeData(
+    lsp4SampleMetadata,
+    LSP4DigitalAssetSchema,
+  );
+
+  // Set up the token contract
+  const token = await ethers.getContractAt('MyCustomToken', customTokenAddress);
+
+  // Get the ERC725Y data key of LSP4
+  const metadataKey = ERC725YDataKeys.LSP4['LSP4Metadata'];
+
+  // Create the transaction payload for setting storage data
+  const lsp4StorageBytecode = token.interface.encodeFunctionData('setData', [
+    metadataKey,
+    encodedLSP4Metadata.values[0],
+  ]);
+}
+```
+
+### Create the batch transaction
+
+After the payloads are prepared correctly, you can execute the [`executeBatch()`](../../contracts/contracts/ERC725/ERC725.md#executebatch) function on the Universal Profile. On the first call, you have to set the transaction target to `0x0`, as the token contract will be initialized. The second call will then use the [previously generated contract address](#prepare-the-transaction-payloads) from the `staticCall` in order to set the metadata.
+
+```ts title="scripts/deployTokenWithMetadataAsUP.ts"
+async function deployTokenWithMetadata() {
+  // Previous code ...
+
+  // Deploy the contract by the Universal Profile
+  const tx = await universalProfile.executeBatch(
+    [
+      // Array of Operation types
+      1, // Operation type: CREATE (Contract deployment)
+      0, // Operation type: CALL (Set storage key on contract)
+    ],
+    [
+      ethers.ZeroAddress, // 0x0 as contract will be initialized
+      customTokenAddress, // Contract address after deployment
+    ],
+    [0, 0], // Value is empty for both operations
+    [
+      tokenBytecodeWithConstructor, // Payload for contract deployment
+      lsp4StorageBytecode, // Payload for setting a data key on the deployed contract
+    ],
+  );
+
+  // Wait for the transaction to be included in a block
+  await tx.wait();
+  console.log('Token deployed at: ', customTokenAddress);
+}
+```
+
+Afterwards, the script is ready to be deployed:
+
+```bash
+npx hardhat --network luksoTestnet run scripts/deployTokenWithMetadataAsUP.ts
+```
+
+:::caution Contract Compilation
+
+Make sure that you [successfully compiled your contract](../smart-contract-developers/getting-started.md) before executing the deployment script.
+
+:::
+
+The transaction will be signed using the configured controller key. After the deployment, you will be able to check the contract on the [Execution Explorer](https://explorer.execution.testnet.lukso.network/).
```

Comment:
The signer will be coming from the Hardhat repository. This guide was initially placed within `/learn/smart-contract-developers`. Hugo moved it out as setting metadata during deployment should be more for experts. The playground repo is still linked and mentioned in the section above, stating that this will be an example using the LSP7 Token Deployment.